### PR TITLE
Don't assume clientDefaultValue is a string

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Look at the parameter's schema type to figure out how to handle it.
Most common case other than string is a constant value.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/autorest.go/issues/651

File: test/maps/azalias/zz_generated_connection.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		azcore.NewTelemetryPolicy(options.telemetryOptions()),
+	}
+	policies = append(policies, options.PerCallPolicies...)
+	policies = append(policies, azcore.NewRetryPolicy(&options.Retry))
+	policies = append(policies, options.PerRetryPolicies...)
+	policies = append(policies, cred.AuthenticationPolicy(azcore.AuthenticationPolicyOptions{Options: azcore.TokenRequestOptions{}}))
+	policies = append(policies, azcore.NewLogPolicy(&options.Logging))
+	hostURL := "https://{geography}.atlas.microsoft.com"
+	if geography == nil {
+		defaultValue := GeographyUs
```
Comment (jhendrixMSFT) [inline test/maps/azalias/zz_generated_connection.go:67]: This is part of the fix.

File: test/maps/azalias/zz_generated_connection.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		geography = &defaultValue
+	}
+	hostURL = strings.ReplaceAll(hostURL, "{geography}", string(*geography))
```
Comment (jhendrixMSFT) [inline test/maps/azalias/zz_generated_connection.go:70]: This is the other part.

File: src/generator/connection.ts [hunk 3]
```diff
@@ -192,11 +192,23 @@ function generateContent(session: Session<CodeModel>, imports: ImportManager): s
...
+        text += `\thostURL = strings.ReplaceAll(hostURL, "{${hostParam.language.go!.serializedName}}", `;
+        switch (hostParam.schema.type) {
+          case SchemaType.Choice:
+          case SchemaType.SealedChoice:
+            text += `string(${pointer}${hostParam.language.go!.name}))\n`;
+            break;
+          case SchemaType.String:
+            text += `${pointer}${hostParam.language.go!.name})\n`;
+            break;
+          default:
```
Comment (chamons) [inline src/generator/connection.ts:208]: Do we know we handle other values here correctly, or should we die/error if we hit a case we don't know about?

**Review by chamons — APPROVED** [review]

File: src/generator/connection.ts (hunk 3 above)
Comment (jhendrixMSFT) [reply src/generator/connection.ts:208]: Given that this is for constructing the host URL, it seems highly unlikely that we'd ever hit this clause (I had debated making this clause throw an error but decided the code generator shouldn't be so "opinionated" here).  However, if it was something else like a number it should work.

**Pull request merged**